ggshield/scan/scannable.py

```diff
@@ -400,21 +415,22 @@ def _collect_results(
     def scan(
         self,
         files: Iterable[File],
-        on_file_chunk_scanned: Callable[[List[File]], None],
+        progress_callback: Callable[..., None],
+        progress_advance: Optional[int] = None,
         scan_threads: int = 4,
     ) -> Results:
 
         with concurrent.futures.ThreadPoolExecutor(
             max_workers=scan_threads, thread_name_prefix="content_scan"
         ) as executor:
-            chunks_for_futures, skipped_chunk = self._start_scans(executor, files)
-
-            if skipped_chunk:
-                # Report skipped_chunk as progress. If we only report files really sent
-                # for scanning then the progress bar won't reach 100%.
-                on_file_chunk_scanned(skipped_chunk)
-
-            return self._collect_results(chunks_for_futures, on_file_chunk_scanned)
+            chunks_for_futures, _ = self._start_scans(
```

Comment:
Since we don't need the `skipped_chunk` part of the returned tuple, `_start_scans()` can be simplified to only return `chunks_for_futures`.

Reply:
Actually, I wanted to keep it as we may want to be more verbose about these `skipped_chunks` in the future. WDYT ?

Reply:
If we decide to do something about it, it will be easy to re-add it. I would remove it to simplify the code.